### PR TITLE
NeMo-UX: add Hf's AutoModelForImageTextToText

### PR DESCRIPTION
**Pull request by akoumpa**
# What does this PR do ?

Adding AutoModelForImageTextToText and PeFT example for instruct models.

Tested with:
- Llama-3.2-11B-Vision-Instruct
- Qwen/Qwen2-VL-2B-Instruct

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/llm/peft/lora.py [hunk 1]
```diff
@@ -112,6 +120,7 @@ class LoRA(PEFT):
         dropout_position (Literal['pre', 'post'], optional): Position for applying dropout.
             Can be 'pre' (before the low-rank projection) or 'post' (after). Defaults to 'pre'.
         a2a_experimental (bool): Enables the experimental All-to-All (A2A) communication strategy. Defaults to False.
+        lora_drype (torch.dtype): Parameter data type for LoRA weights. Default None (will use model's dtype).
```
Comment (yaoyu-33) [inline nemo/collections/llm/peft/lora.py:123]: why this change is needed?

Comment (akoumpa) [reply nemo/collections/llm/peft/lora.py:123]: HF supports loading 4bit weights, the adapter will use the model's dtype (in this case 4bit) which is untrainable -> therefore you have to be able to pass the dtype manually.

File: tests/collections/vlm/hf/peft.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        batch["pixel_values"] = batch["pixel_values"].to(torch.bfloat16)
+
+        labels = batch["input_ids"].clone()
+        labels[torch.isin(labels, skipped_tokens)] = -100
+        batch["labels"] = labels
+        return batch
+
+    return vlm.HFDatasetDataModule(
+        DATA_PATH,
+        split="train[:10]",
```
Comment (yaoyu-33) [inline tests/collections/vlm/hf/peft.py:69]: what does [:10] mean here?

File: nemo/collections/vlm/hf/model/hf_auto_model_for_image_text_to_text.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    if mask is not None:
+        loss = F.cross_entropy(logits, targets, reduction='none')
+        return torch.mean(loss[mask == 1])
+    else:
+        return F.cross_entropy(logits, targets)
+
+
+def align_labels(logits, labels):
+    """Aligns labels to logits
+    if length of labels == logits, then will remove first item
```
Comment (yaoyu-33) [inline nemo/collections/vlm/hf/model/hf_auto_model_for_image_text_to_text.py:36]: tokens 0 1 2 3
labels 0 1 2 3
logits 1 2 3 4.

in regular vlm training, we provide tokens as 0 1 2, and labels as 1 2 3, the requirements seems to be a bit different here.

File: tests/collections/vlm/hf/peft.py (hunk 2 above)
Comment (akoumpa) [reply tests/collections/vlm/hf/peft.py:69]: just use the first ten items, similar to python list operator

**Review by yaoyu-33 — DISMISSED** [review]

**Review by yaoyu-33 — DISMISSED** [review]

**Review by pablo-garay — APPROVED** [review]
